### PR TITLE
Refactor StarsRating

### PR DESCRIPTION
**Pull request by Cygap**
# Description
Refactored stars rating to be more reusable and have text as a prop. Show two StarsRatings if user is logged in. Also it is ready for getting user.rating info, which we will have later.
# Jira task link
https://hackyourfuture-dk.atlassian.net/browse/VIEW-71
# Screenshots (if applicable)
image

### CODE REVIEW

File: packages/client/src/components/BigMovieCard/BigMovieCard.jsx [hunk 1]
```diff
@@ -48,6 +57,12 @@ export const BigMovieCard = ({ currentMovie }) => {
     borderRadius: '1.25rem',
     margin: '2.5rem 0',
   });
+  // eslint-disable-next-line no-console
```
Comment (mohamedaldahoul) [inline packages/client/src/components/BigMovieCard/BigMovieCard.jsx:60]: Can you remove the comment and console.log, pls?

File: packages/client/src/components/BigMovieCard/BigMovieCard.jsx [hunk 2]
```diff
@@ -111,14 +126,36 @@ export const BigMovieCard = ({ currentMovie }) => {
             alignSelf: 'flex-end',
           }}
         >
+          {user ? (
```
Comment (mohamedaldahoul) [inline packages/client/src/components/BigMovieCard/BigMovieCard.jsx:129]:
```suggestion
          {user  && (
```
then remove the ""

File: packages/client/src/components/BigMovieCard/BigMovieCard.jsx (hunk 1 above)
Comment (Cygap) [reply packages/client/src/components/BigMovieCard/BigMovieCard.jsx]: 🤦🏻‍♂️thanks for spotting this. Already removed it.

File: packages/client/src/components/BigMovieCard/BigMovieCard.jsx (hunk 2 above)
Comment (Cygap) [reply packages/client/src/components/BigMovieCard/BigMovieCard.jsx]: Thanks for the suggestion, it looks more concise now!

**Review by magdazelena — APPROVED** [review]
👏

File: packages/client/src/components/BigMovieCard/BigMovieCard.jsx [hunk 3]
```diff
@@ -111,14 +120,34 @@ export const BigMovieCard = ({ currentMovie }) => {
...
+                : 'No rating given'
+            }
+            rating={currentMovie.rating}
             alignSelf="flex-end"
           />
-
+          <ReviewDialog
+            initialState={open}
+            handleClose={(event, value) =>
+              handleOpenReview(new Event('Review submitted'), value)
```
[the remark this answers is not in the document]
Comment (Cygap) [reply packages/client/src/components/BigMovieCard/BigMovieCard.jsx]: Looked through the code - that Event was not needed at all :) Thanks for the remark. I removed it

File: packages/client/src/components/BigMovieCard/BigMovieCard.jsx [hunk 4]
```diff
@@ -111,14 +120,34 @@ export const BigMovieCard = ({ currentMovie }) => {
...
+                  } review${currentMovie.number_of_ratings === 1 ? '' : 's'}`
+                : 'No rating given'
+            }
+            rating={currentMovie.rating}
             alignSelf="flex-end"
           />
-
+          <ReviewDialog
+            initialState={open}
+            handleClose={(event, value) =>
```
[the remark this answers is not in the document]
Comment (Cygap) [reply packages/client/src/components/BigMovieCard/BigMovieCard.jsx]: Renamed all the occurrences of the unused event to _event. Thank for sharing the best practice.

**Comment by Cygap** [comment]
Addressed most recent comments, synchronized with the updated main. Resolved conflicts.

**Pull request merged**